rust/datafusion/src/sql/planner.rs

```diff
@@ -1305,6 +1304,74 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             )))
         }
     }
+
+    fn show_columns_to_plan(
+        &self,
+        extended: bool,
+        full: bool,
+        table_name: &ObjectName,
+        filter: Option<&ShowStatementFilter>,
+    ) -> Result<LogicalPlan> {
+        if filter.is_some() {
+            return Err(DataFusionError::Plan(
+                "SHOW COLUMNS with WHERE or LIKE is not supported".to_string(),
+            ));
+        }
+
+        if !self.has_table("information_schema", "columns") {
+            return Err(DataFusionError::Plan(
+                "SHOW COLUMNS is not supported unless information_schema is enabled"
+                    .to_string(),
+            ));
+        }
+
+        if self
+            .schema_provider
+            .get_table_provider(table_name.try_into()?)
+            .is_none()
+        {
+            return Err(DataFusionError::Plan(format!(
+                "Unknown relation for SHOW COLUMNS: {}",
+                table_name
+            )));
+        }
+
+        // Figure out the where clause
+        let columns = vec!["table_name", "table_schema", "table_catalog"].into_iter();
+        let where_clause = table_name
+            .0
+            .iter()
+            .rev()
```

Comment:
Why `rev` and not listing the items in column in reverse?

Reply:
I was thinking because the identifier may not have a `table_catalog` or `table_schema` 

So you have to handle the case of 

`table_name` (position 0)
`table_name` (position 1), `schema_name` (postition 0)
`table_name` (postition 2), `schema_name` (postition 1), `catalog_name` (position 0), 

So this formulation was what I could come up with that would match them up to the information_schema column names

I am open to other ways of doing it as well if you have suggestions

Reply:
You are totally right 👍 I don't have a suggestion for an easier way